.github/workflows/test-changed-firestore.yml

```diff
@@ -29,16 +33,104 @@ jobs:
           yarn
       - name: build
         run: yarn build:changed firestore
-      - name: Run tests if firestore or its dependencies has changed
-        run: yarn test:changed firestore
+      - name: Archive build
+        if: ${{ !cancelled() }}
+        run: |
+          tar -cf build.tar .
```

Comment:
It may be a good idea to put the output file of `tar` into a some other directory that is not being included in the archive so that the partially-completed tar file being created by tar itself isn't included in the archive.